enclave_wrangler/objects_api.py

```diff
@@ -405,7 +405,15 @@ def fetch_cset_and_member_objects(
 
     return {'OMOPConceptSetContainer': cset_containers, 'OMOPConceptSet': cset_versions,
             'OmopConceptSetVersionItem': flat_expression_items, 'OMOPConcept': flat_member_items}
-
+def is_new_db_data(since: Union[datetime, str]):
+    "Returns True if there is new data for the db, false if not"
```

Comment:
I created a new function to test for new data, I'm not sure this file is the best place for it.

Reply:
It's OK to be there right now. I'm going to be moving a lot of stuff from inside this file into `backend/db/` instead, at some point soon probably. I can move this at that time.

I'm supposing that this function will call the first part of `fetch_cset_and_member_objects()`. This function returns this structure of new stuff to put in our DB (copy/pasting from its docstring):

```
    :return
      - cset containers
      - cset versions
          - member items
          - expression items
      - member items
      - expression items
```

Right now the way it's set up, it fetches cset versions and, based on that, it fetches all of these other things as they relate to these cset versions. Siggie wants me to change it to fetch containers independently of cset versions, so I'll be changing that soon.

Anyway, let's say that it tries to fetch new cset versions since `since`, but finds nothing. I believe the way that it will turn out is that this:
```py
    return {'OMOPConceptSetContainer': cset_containers, 'OMOPConceptSet': cset_versions,
            'OmopConceptSetVersionItem': flat_expression_items, 'OMOPConcept': flat_member_items}
```

will resolve to this:
```py
    return {'OMOPConceptSetContainer': [], 'OMOPConceptSet': [],
            'OmopConceptSetVersionItem': [], 'OMOPConcept': []}
```

Honestly though, if there's no new data, I should just have it return `None`.

In any case, I guess what I'm thinking is that you can have `is_new_db_data()` simply call `fetch_cset_and_member_objects()` and return true or false based on what it sees in response. But another alternative is that we can just skip `is_new_db_data()` and use `fetch_cset_and_member_objects()` in its place. It kinda depends on why this function is being created / what it will be used for. It wouldn't be performant to fetch a bunch of stuff, just to see if there is any data in the DB, just fo tech it again.

Although, another thing you could do is to tell if there's new data is to just call a subset of `fetch_cset_and_member_objects()`. As I was mentioning earlier, it checks for cset versions and then fetches other stuff based on that. And soon will be changed to also fetch containers whether or not there are any new cset versions. So your function could just fetch those two things independently and return true/false depending on if anything comes back:

```py
cset_versions: List[Dict] = fetch_objects_since_datetime('OMOPConceptSet', since)
cset_containers: List[Dict] = fetch_objects_since_datetime('OMOPConceptSetContainer', since)
return True if cset_versions or cset_containers else False
```

Then again, maybe there is some other way to check if there's new data for us in the enclave other than what I've set up now, like a specific endpoint in the enclave API, but I haven't come across such alternative AFAIK.

Reply:
New function was removed based on other stuff I said.

Reply:
I actually changed it to return none in fetch_cset_and_members based on the the two lines of code you wrote, then I changed all functions that call fetch_cset_and_members to be able to handle that. 

```python
cset_versions: List[Dict] = fetch_objects_since_datetime('OMOPConceptSet', since)
cset_containers: List[Dict] = fetch_objects_since_datetime('OMOPConceptSetContainer', since) ```

Reply:
Confirming that `None` is being returned in the correct way in `fetch_cset_and_member_objects()`, thanks!
```py
    if not cset_containers and not cset_versions:
        return None
```

You also correctly made this change to `csets_and_members_enclave_to_db()`:
```py
   if not csets_and_members:
        return None
```

This one is a mistake, though!
```py
def fetch_all_new_objects():
    csets_and_members: Dict[str, List] = fetch_cset_and_member_objects(since)
    if not csets_and_members:
        return None
    researchers = get_researchers()
    projects = get_projects()
```

I think it's a good thing that you looked at where `fetch_cset_and_member_objects()` was being called, and made these changes. 
And it is not a big deal that this is a mistake, because we're not even using this function `fetch_all_new_objects()` yet. Right now we're going from `refresh_db()` -> `csets_and_members_enclave_to_db()` -> `fetch_cset_and_member_objects()`
But eventually when we get to #341 we will go from `refresh_db()` -> `all_new_objects_enclave_to_db()` -> `fetch_all_new_objects()`, which will also call -> `fetch_cset_and_member_objects()`
And the reason it's not correct in this function to return if no `csets_and_members`, is just because there are no new concept sets, etc, from that function, that does not mean that there is not new researcher and project data.
I went ahead and fixed this and pushed so you don't need to worry about on your end!